### PR TITLE
Be more accepting of select options

### PR DESCRIPTION
**Pull request by Janpot**
Make sure we always show something, even if the options array contains numbers, or `undefined`

### CODE REVIEW

**Review by apedroferreira — CHANGES_REQUESTED** [review]
Works well, just added some thoughts about the tests! Feel free to disagree.

File: test/integration/components/index.spec.ts [hunk 1]
```diff
@@ -1,21 +1,34 @@
...
+async function waitForComponents(page: Page, frame: Page | FrameLocator = page, isEditor = false) {
+  await frame.locator('text="foo button"').waitFor({ state: 'visible' });
+  await frame.locator('img[alt="foo image"]').waitFor({ state: 'attached' });
+  await frame.locator('text="foo datagrid column"').waitFor({ state: 'visible' });
+  await frame.locator('text="custom component 1"').waitFor({ state: 'visible' });
+  await frame.locator('label:has-text("foo textfield")').waitFor({ state: 'visible' });
+  await frame.locator('text="foo typography"').waitFor({ state: 'visible' });
+  await frame.locator('label:has-text("foo select")').waitFor({ state: 'visible' });
+
+  const optionsSelect = frame.getByRole('button', { name: /select with options/ });
```
[the remark this answers is not in the document]
Comment (Janpot) [reply test/integration/components/index.spec.ts:18]: Maybe, I'm sort of avoiding splitting tests too much as to keep the footprint down. The whole test with setup and page load takes about 2/3 seconds. While just interacting with this select takes an order of magnitude less.
But perhaps it is sensible to add one test per component

File: test/integration/components/index.spec.ts [hunk 2]
```diff
@@ -1,21 +1,34 @@
...
-async function waitForComponents(page: Page | FrameLocator) {
-  await page.locator('text="foo button"').waitFor({ state: 'visible' });
-  await page.locator('img[alt="foo image"]').waitFor({ state: 'attached' });
-  await page.locator('text="foo datagrid column"').waitFor({ state: 'visible' });
-  await page.locator('text="custom component 1"').waitFor({ state: 'visible' });
-  await page.locator('label:has-text("foo textfield")').waitFor({ state: 'visible' });
-  await page.locator('text="foo typography"').waitFor({ state: 'visible' });
-  await page.locator('label:has-text("foo select")').waitFor({ state: 'visible' });
+async function waitForComponents(page: Page, frame: Page | FrameLocator = page, isEditor = false) {
+  await frame.locator('text="foo button"').waitFor({ state: 'visible' });
```
[the remark this answers is not in the document]
Comment (Janpot) [reply test/integration/components/index.spec.ts]: I'll assign those to variables with more descriptive names. That makes it self-documenting.

File: test/integration/components/index.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (apedroferreira) [reply test/integration/components/index.spec.ts]: Yeah that's right, with a page load in every test things can get a lot slower...
One test per component sounds like a good future-proof approach - we could also maybe slowly get rid of this general test as we create one test for each component.
I think the way you changed it is good!

**Review by apedroferreira — APPROVED** [review]

**Pull request merged**